src/java/org/apache/cassandra/index/sai/disk/v3/CassandraDiskAnn.java

```diff
@@ -112,54 +108,21 @@ public int size()
     }
 
     /**
-     * @return Row IDs associated with the topK vectors near the query
-     *
      * @param queryVector the query vector
      * @param topK the number of results to look for in the index (>= limit)
-     * @param limit the maximum number of results to return
+     * @param threshold the minimum similarity score to accept
      * @param acceptBits a Bits indicating which row IDs are acceptable, or null if no constraints
      * @param context unused (vestige from HNSW, retained in signature to allow calling both easily)
-     */
-    @Override
-    public VectorPostingList search(float[] queryVector, int topK, int limit, Bits acceptBits, QueryContext context)
-    {
-        return search(queryVector, topK, 0, limit, acceptBits, context);
-    }
-
-    /**
+     * @param nodesVisitedConsumer a consumer that will be called with the number of nodes visited during the search
```

Comment:
"... May be invoked multiple times as we dig deeper into the graph to find more candidates when we discover a shadowed row in the results."

Reply:
It's actually only called once when we close the `AutoResumingNodeScoreIterator`

Reply:
In that case I may object to the consumer pattern more strongly :)